### ISSUE TITLE
potential function gives wrong gradient under JIT for CUDA tensor (PyTorch issue)

### ISSUE THREAD

**Issue by fehiepsi** [pyro-ppl/pyro#1955]
This issue is detected in #1954. A simple script to replicate the issue

```
import torch
import pyro
import pyro.distributions as dist
from pyro.infer.mcmc.util import initialize_model
from pyro.ops.integrator import potential_grad

def model(data):
    z = pyro.sample("z", dist.Normal(0, 1))
    pyro.sample("obs", dist.Normal(0, z), obs=data)

def get_grad(jit=True, cuda=True):
    pyro.set_rng_seed(0)
    torch.set_default_tensor_type(torch.cuda.FloatTensor if cuda else torch.FloatTensor)
    data = torch.tensor([1., 2.])
    _, potential_fn, _, _ = initialize_model(model, (data,), jit_compile=jit)
    return potential_grad(potential_fn, {'z': torch.tensor(1.)})

get_grad(True), get_grad(False), get_grad(True, False), get_grad(False, False)
```
which returns different results for jit and non-jit
```
(({'z': tensor(-7., device='cuda:0')}, tensor(5.7568, device='cuda:0')),
 ({'z': tensor(-2., device='cuda:0')}, tensor(5.7568, device='cuda:0')),
 ({'z': tensor(-2.)}, tensor(5.7568)),
 ({'z': tensor(-2.)}, tensor(5.7568)))
```

**Comment by fehiepsi** [comment]
Actually, this seems like a PyTorch issue
```
def f(z):
    return -dist.Normal(0, z).log_prob(torch.tensor([1., 2.])).sum()

torch.set_default_tensor_type(torch.cuda.FloatTensor)
jf = torch.jit.trace(f, torch.tensor(1.))

z = torch.tensor(1., requires_grad=True)
y = jf(z)
print(torch.autograd.grad(y, (z,))[0])

z = torch.tensor(1., requires_grad=True)
y = f(z)
print(torch.autograd.grad(y, (z,))[0])
```
which returns
```
tensor(-8.)
tensor(-3.)
```

**Comment by fehiepsi** [comment]
This issue is fixed in PyTorch latest version.
